docs/source/design/composite-time-series.rst

```diff
@@ -0,0 +1,379 @@
+
+However, on request for the timeseries the list of composite time series is consulted and used if present, otherwise passthrough to normal
+time series retrieval.
+
+
+Composite Time Series Definition
+================================
+
+.. code-block::jsonc
+
+    {
+    "office": "<string>",   
+    "name": "<ts id name>",
+    "is-authoritative": true|false,
+    "seasonal-information": "<office>/<location level>", // Optional, reference to location level that returns 0 for out-of-service 
+                                                         // or 1 for in-service
+    "members": [
+        {
+            "time-series-id": "TS ID for this range",
+            "start": "start date of this", // Inclusive
+            "end": "end date of this range", // Exclusive, can be null
+            "notes": "text",            
+        }
+    ]
+    // array above *should* be sorted by start when provided to user.
+    }
+
+
+Operations required:
+
+* Create
+* Remove member (ts id + range)
+* Add member
+* List members
+* Replace all members?
+* Update member
+* Delete
+
+Immutable fields:
+
+Fields marked immutable above cannot be updated. At this time no field are thought to be immutable.
+
+
+
+Composite Time Series Response
+==============================
+
+.. code-block::jsonc
+
+    {
+    // ... as current TimeSeries JSON
+    "composite-members-present": [
+        // member definition from above
+    ] 
+    }
+
+
+Supported Operations:
+
+* Get, through existing TimeSeries classes.
+
+
+Storage of member information
+================================
+
+#. Store in Clob as we refine the design - cache appropriately to avoid any major performance issues.
+#. Create appropriate tables once the design is stable - still cache things.
+
+System responsibility for "knowing" to process composite.
+=========================================================
+
+Time Series Catalog
+-------------------
+
+Time Series Catalog should show composite time series and allow searching by "authoritative"
+
+TimeSeries DTO
+--------------
+
+Add nullable "members" property.
+
+TimeSeriesDao
+-------------
+
+If the system sees the "Composite" marker/determines is composite retrieve the members for the range and build the time series.
+
+.. NOTE:: 
+    Considering the user may request the *entire* Period-of-record,
+    start the retrieval in a job queue, and return a status URL to the user for future download. I have see such mechanism 
+    for bulk data in other systems. Maybe return an "I'm working on it variant" that the controller can know how to format.
+
+    Perhaps we do this for data beyond "x amount"?
+
+Error handling and other conditions.
+====================================
+
+Versioned (date) time series
+----------------------------
+
+It is an error to specify a Version (date) when requesting composite data.
```

Comment:
Meaning composite time series always represent the latest version date for all member time series?

Reply:
I added that to the text.